### PR TITLE
test(amazon-cognito-identity-js): add unit tests for CognitoUser and extend constants

### PR DESCRIPTION
**Pull request by nickarocho**
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazon-cognito-identity-js/__tests__/CognitoUser.test.js [hunk 1]
```diff
@@ -44,3 +53,178 @@ describe('CognitoUser constructor', () => {
...
+		onFailure: jest.fn(),
+		onSuccess: jest.fn(),
+		customChallenge: jest.fn(),
+	};
+
+	afterAll(() => {
+		jest.restoreAllMocks();
+	});
+
+	afterEach(() => {
```
Comment (evcodes) [inline packages/amazon-cognito-identity-js/__tests__/CognitoUser.test.js:102]: why do you do this here? Maybe we can discuss this, I'd like to know how to use the afterEach and callback in this way

File: packages/amazon-cognito-identity-js/__tests__/CognitoUser.test.js [hunk 2]
```diff
@@ -44,3 +53,178 @@ describe('CognitoUser constructor', () => {
...
+		const user = new CognitoUser({
+			Username: 'username',
+			Pool: cognitoUserPool,
+		});
+		const authDetails = new AuthenticationDetails(
+			authDetailDataWithValidationData
+		);
+
+		user.initiateAuth(authDetails, callback);
+		// expect(authDetails.getAuthParameters().USERNAME).toEqual(user.username);
```
Comment (evcodes) [inline packages/amazon-cognito-identity-js/__tests__/CognitoUser.test.js:178]: is this supposed to be commented?

**Review by evcodes — COMMENTED** [review]
no hard blockers, just small details. LGTM

File: packages/amazon-cognito-identity-js/__tests__/CognitoUser.test.js (hunk 1 above)
Comment (nickarocho) [reply packages/amazon-cognito-identity-js/__tests__/CognitoUser.test.js:102]: I am mocking a few different functions per test, so it's just a lazy way to clear the mocks after each test.

**Pull request merged**